File: stack_linked_list.c
//Implementation of stacks using linked list.
#include <stdio.h>
#include <stdlib.h>
void push(int );
int pop();
int peek();
void print();
struct node
{
	int data;
	struct node* link;
}*top=NULL;
int isempty()
{
	if(top==NULL){
		return 1;
	}
	else{
		return 0;
	}
}
int peek()
{
	if(top==NULL)
	{
		printf("Stack Underflow\n");
		exit(1);
	}
	else{
		return top->data;
	}
}
int pop()
{
	struct node* temp;
	int val;
	if(isempty()){
		printf("Stack Underflow\n");
		exit(1);
	}
	temp=top;
	val=temp->data;
	top=top->link;
	free(temp);
	temp=NULL;
	return val;
	
}
void push(int data)
{
	struct node* newnode;
	newnode=malloc(sizeof(newnode));
	if(newnode==NULL)
	{
		printf("Stack Overflow");
		exit(1);
	}
	newnode->data=data;
	newnode->link=NULL;
	newnode->link=top;
	top=newnode;
}
void print()
{
	struct node* temp;
	temp=top;
	if(isempty()){
		printf("Stack Underflow\n");
		exit(1);
	}
	printf("The stack elements are:\n");
	while(temp){
		printf("%d ",temp->data);
		temp=temp->link;
	}
	printf("\n");
}
int main()
{
	int choice,data;
	while(1)
	{
		printf("Enter 1 for push\n");
		printf("Enter 2 for pop\n");
		printf("Enter 3 to see peek\n");
		printf("Enter 4 for print\n");
		printf("Enter 5 for quit\n");
		printf("Enter your choice\n");
		scanf("%d",&choice);
		switch(choice){
			case 1:
				printf("Enter element to be pushed\n");
				scanf("%d",&data);
				push(data);
				break;
			case 2:
				data=pop();
				printf("Deleted element=%d\n",data);
				break;
			case 3:
				data=peek();
				printf("Peek element=%d",data);
				break;
			case 4:
				print();
				break;
			case 5:
				exit(1);
				break;
			default:
				printf("Invalid choice\n");
		}
	}
	return 0;
}

File: Stack_Linked_List.java

public class Stack_Linked_List {
    public static class Node{
        int data;
        Node next;
        Node(int d)
        {
            data=d;
            next=null;
        }
    }
    public static class Stack{
        public Node head=null;
        public boolean isEmpty()
        {
            return head==null;
        }
        public void push(int d)
        {
            Node newNode=new Node(d);
            if(head==null)
            {
                head=newNode;
                return;
            }
            newNode.next=head;
            head=newNode;
        }
        public int pop()
        {
            if(isEmpty()){
                return -1;
            }
            Node top=head;
            head=head.next;
            return top.data;
        }
        public int peek()
        {
            if(isEmpty())
            {
                return -1;
            }
            Node top=head;
            return top.data;
        }
    }
    public static void main(String args[])
    {
        Stack ob=new Stack();
        ob.push(1);
        ob.push(2);
        ob.push(3);
        ob.push(4);
        while(!ob.isEmpty())
        {
            System.out.println(ob.peek());
            ob.pop();
        }

    }
}
